src/chapters/exceptions/exceptions-control-flow.rst

```diff
@@ -1,71 +1,137 @@
+.. tip:: ``catch`` blocks only execute if an exception is thrown
 
-JavaScript also provides us with a ``finally`` statement, a block of code that always runs every time regardless if the ``try``, or ``catch`` blocks run.
+
+Finally
+-------
+.. index:: ! finally
+
+.. index::
+   single: try catch; finally
+
+JavaScript also provides a ``finally`` block which must be used with ``try`` and ``catch`` blocks.  A ``finally`` block code runs after the ``try`` and ``catch``.
+What is special about ``finally`` is that ``finally`` is ALWAYS ran, even if an exception is NOT thrown.
 
 .. admonition:: Example
 
-   .. sourcecode:: js
+   Let's update the above example to print out the index the user entered. Importantly we want this message to be printed EVERY time the code runs.
+   Notice the ``console.log`` statement on line 11.
+
+   .. replit:: js
+      :slug: control-flow-type-error-finally
       :linenos:
 
+      const input = require('readline-sync');
+
+      let animals = [{name: 'cat'}, {name: 'dog'}];
+      let index = Number(input.question("Enter index of animal:"));
+
       try {
-          console.log(x[0]);
-      }
-      catch(ReferenceError) {
-          console.log("We caught a ReferenceError, but our program continues to run!");
-          console.log("0");
-      }
-      finally {
-          console.log("In the finally statement!");
+         console.log('animal at index:', animals[index].name);
+      } catch(TypeError) {
+         console.log("We caught a TypeError, but our program continues to run!");
+      } finally {
+         console.log("You tried to access an animal at index:", index);
       }
 
+      console.log("the code goes on...");
+
    **Console Output**
 
+   If the user enters ``7``:
+   ::
+
+      Enter index of animal: 7
+      We caught a TypeError, but our program continues to run!
+      You tried to access an animal at index: 7
+      the code goes on...
+
+   If the user enters ``1``:
    ::
 
-      We caught a ReferenceError, but our program continues to run!
-      0
-      In the finally statement!
+      Enter index of animal: 1
+      animal at index: dog
+      You tried to access an animal at index: 1
+      the code goes on...
 
-   This try...catch block is identical to the example above, except this time we have a ``finally`` statement. The finally statement runs upon the conclusion of the try...catch statements. It is often used to remove hanging resources.
-     
-Errors are a normal part of coding. JavaScript defines the most common errors and provides messages to help us debug. We can extend the functionality around exceptions to further assist us in debugging or to handle expected issues in our code.
 
 Check Your Understanding
 ------------------------
 
 .. admonition:: Question
 
    What statement do we use if we want to attempt to run code, but think an
-   exception might be raised?
+   exception might be thrown?
```

Comment:
Since one of the options is ``throw``, then the original "raised" might be a better word to use here.

Reply:
I feel it's too confusing to use the term raised. I'm trying to beat the term thrown into their heads.